### PR TITLE
chore: move printing errors to `nargo_cli`

### PR DESCRIPTION
**Pull request by f01dab1e**
# Description

Move printing errors to `nargo_cli`

## Problem

Resolves #1530 

# PR Checklist

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by f01dab1e** [comment]
@kevaundray, what do you think about it?

File: crates/nargo_cli/src/errors.rs [hunk 1]
```diff
@@ -45,7 +45,7 @@ pub(crate) enum CliError<B: Backend> {
 
     /// Error while compiling Noir into ACIR.
     #[error("Failed to compile circuit")]
-    CompilationError,
+    CompilationError(#[from] noirc_errors::Error),
```
Comment (kevaundray) [inline crates/nargo_cli/src/errors.rs:48]: How does this work if a program returns more than one error?

Comment (f01dab1e) [reply crates/nargo_cli/src/errors.rs:48]: There will be problems with this, but maybe we can go and do something like this:
https://github.com/noir-lang/noir/blob/053dba0ccc18757aef8e12fa170103a46de07244/crates/noirc_frontend/src/hir/type_check/errors.rs#L41

**Comment by phated** [comment]
@f01dab1e I think we're going to supersede this PR with https://github.com/noir-lang/noir/pull/1598 since we need to get back a vector of diagnostics to use in the LSP. I was wondering if you could look into #1599 since you've been digging into this code and I took the quick-and-dirty route of adding the bad API 🙈

**Pull request closed without merging**